File: java-spring/kafka/src/main/java/com/example/kafka/domain/Feed.java
package com.example.kafka.domain;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@ToString
@NoArgsConstructor
public class Feed {

    private String title;
    private String content;

    @Builder
    public Feed(String title, String content) {
        this.title = title;
        this.content = content;
    }
}

File: kotlin-spring/feed/src/main/kotlin/com/example/feed/domain/feed/domain/Feed.kt
package com.example.feed.domain.feed.domain

import com.example.feed.domain.user.domain.User
import com.example.feed.global.entity.BaseEntity
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

@Entity
class Feed(

        title: String,
        content: String,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "user_id", nullable = false)
        val user: User,

): BaseEntity() {

        @Column(columnDefinition = "VARCHAR(30)", nullable = false)
        var title: String = title
                protected set

        @Column(columnDefinition = "VARCHAR(1000)", nullable = false)
        var content: String = content
                protected set

        fun update(title: String, content: String) {
                this.title = title
                this.content = content
        }
}